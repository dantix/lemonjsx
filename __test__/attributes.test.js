import h from '../src/index';

describe('attributes', () => {
  describe('general', () => {
    it('should not create section with empty attributes', () => {
      expect(<section />).toMatchSnapshot();
    });
  });

  describe('alignment', () => {
    it('should pass thorugh proper value', () => {
      expect(<section align="r" />).toMatchSnapshot();
    });

    it('should omit invalid value', () => {
      expect(<section align="right" />).toMatchSnapshot();
    });
  });

  describe('colors', () => {
    it('should pass thorugh #rgb value', () => {
      const section = (
        <section
          background="#fff"
          foreground="#fff"
          underline="#000"
        />
      );

      expect(section).toMatchSnapshot();
    });

    it('should pass thorugh #rrggbb value', () => {
      const section = (
        <section
          background="#ffffff"
          foreground="#ffffff"
          underline="#Ff3acb"
        />
      );

      expect(section).toMatchSnapshot();
    });

    it('should pass thorugh #aarrggbb value', () => {
      const section = (
        <section
          background="#ffffffff"
          foreground="#ffffffff"
          underline="#AbCdeF12"
        />
      );

      expect(section).toMatchSnapshot();
    });

    it('should omit invalid color value', () => {
      const section = (
        <section
          background="#1234f"
          foreground="#badcolor"
          underline="#wrongcolor"
        />
      );

      expect(section).toMatchSnapshot();
    });
  });

  describe('font', () => {
    it('should pass through proper value', () => {
      expect(<section font={1} />).toMatchSnapshot();
    });

    it('should pass through proper value', () => {
      expect(<section font="3" />).toMatchSnapshot();
    });

    it('should omit invalid value', () => {
      expect(<section font="Ubuntu" />).toMatchSnapshot();
    });

    it('should omit invalid value', () => {
      expect(<section font={6} />).toMatchSnapshot();
    });
  });

  describe('offset', () => {
    it('should pass through proper value', () => {
      expect(<section offset={150} />).toMatchSnapshot();
    });

    it('should omit invalid value', () => {
      expect(<section offset={-100} />).toMatchSnapshot();
    });

    it('should omit invalid value', () => {
      expect(<section offset="some gibberish" />).toMatchSnapshot();
    });
  });
});
