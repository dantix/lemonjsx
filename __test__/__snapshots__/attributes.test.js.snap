exports[`attributes alignment should omit invalid value 1`] = `""`;

exports[`attributes alignment should pass thorugh proper value 1`] = `"%{r}"`;

exports[`attributes colors should omit invalid color value 1`] = `""`;

exports[`attributes colors should pass thorugh #aarrggbb value 1`] = `"%{B#ffffffff F#ffffffff U#AbCdeF12}"`;

exports[`attributes colors should pass thorugh #rgb value 1`] = `"%{B#fff F#fff U#000}"`;

exports[`attributes colors should pass thorugh #rrggbb value 1`] = `"%{B#ffffff F#ffffff U#Ff3acb}"`;

exports[`attributes font should omit invalid value 1`] = `""`;

exports[`attributes font should omit invalid value 2`] = `""`;

exports[`attributes font should pass through proper value 1`] = `"%{T1}"`;

exports[`attributes font should pass through proper value 2`] = `"%{T3}"`;

exports[`attributes general should not create section with empty attributes 1`] = `""`;
